### PR TITLE
Backend/spec-migration-phase-1/search on_search tags

### PR DESCRIPTION
**Pull request by shreyaxh**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [x] ONDC Spec Migration

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

**Review by NikithShetty — CHANGES_REQUESTED** [review]

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/API/Beckn/Init.hs [hunk 1]
```diff
@@ -44,23 +48,38 @@ handler = init
...
+      withTransactionIdLogTag reqV2 $ do
+        logTagInfo "Init API Flow" "Reached"
+        ACL.buildInitReqV2 subscriber reqV2
+
+  Redis.whenWithLockRedis (initLockKey dInitReq.estimateId) 60 $ do
+    validatedRes <- DInit.validateRequest transporterId dInitReq
+    fork "init request processing" $ do
+      Redis.whenWithLockRedis (initProcessingLockKey dInitReq.estimateId) 60 $ do
+        dInitRes <- DInit.handler transporterId dInitReq validatedRes
+        case req of
```
Comment (NikithShetty) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/API/Beckn/Init.hs:71]: this switch is supposed to happen based on flag and not the request type

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/API/Beckn/Search.hs [hunk 2]
```diff
@@ -44,26 +48,56 @@ search ::
...
+          logTagInfo "SearchV2 API Flow" "Reached"
+          ACL.buildSearchReqV2 transporterId subscriber reqV2
+
+  Redis.whenWithLockRedis (searchLockKey dSearchReq.messageId transporterId.getId) 60 $ do
+    merchant <- DSearch.validateRequest transporterId dSearchReq
+    fork "search request processing" $
+      Redis.whenWithLockRedis (searchProcessingLockKey dSearchReq.messageId transporterId.getId) 60 $ do
+        dSearchRes <- DSearch.handler merchant dSearchReq
+        let callbackUrl = gateway.subscriber_url
+        case req of
```
Comment (NikithShetty) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/API/Beckn/Search.hs:73]: same here, make switch case based on flag and not on request type everywhere else

**Review by NikithShetty — CHANGES_REQUESTED** [review]

File: Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/Confirm.hs [hunk 3]
```diff
@@ -48,7 +48,6 @@ confirm personId quoteId paymentMethodId = SConfirm.confirm SConfirm.DConfirmReq
 -- cancel booking when QUOTE_EXPIRED on bpp side, or other EXTERNAL_API_CALL_ERROR catched
 cancelBooking :: (CacheFlow m r, EncFlow m r, EsqDBFlow m r) => DRB.Booking -> m ()
 cancelBooking booking = do
-  logTagInfo ("BookingId-" <> getId booking.id) ("Cancellation reason " <> show DBCR.ByApplication)
```
[the remark this answers is not in the document]
Comment (shreyaxh) [reply Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/Confirm.hs:51]: Must have been removed accidentally, added it back

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/API/Beckn/Confirm.hs [hunk 4]
```diff
@@ -99,3 +123,12 @@ confirmLockKey id = "Driver:Confirm:BookingId-" <> id
 
 confirmProcessingLockKey :: Text -> Text
 confirmProcessingLockKey id = "Driver:Confirm:Processing:BookingId-" <> id
+
+decodeReq :: MonadFlow m => ByteString -> m (Either Confirm.ConfirmReq Confirm.ConfirmReqV2)
+decodeReq reqBS =
+  case A.eitherDecodeStrict reqBS of
```
[the remark this answers is not in the document]
Comment (shreyaxh) [reply Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/API/Beckn/Confirm.hs:129]: Done for all APIs

**Pull request closed without merging**